replica/table.cc

```diff
@@ -1994,6 +1998,32 @@ static size_t memory_usage_of(const utils::chunked_vector<frozen_mutation_and_sc
     }), size_t{base_overhead_bytes * ms.size()});
 }
 
+// Finds all endpoints to which the view updates will be sent, reads their current view backlogs,
+// chooses the backlog that is the most full and calculates the amount of time to sleep for based on it.
+// The local node's backlog is also included in the calculations, as every view update spawns a local
+// task before being sent away to the remote node.
+static std::chrono::microseconds calculate_throttling_delay_for_view_update_batch(
+    const utils::chunked_vector<frozen_mutation_and_schema>& view_updates,
+    const dht::token& base_token,
+    const shared_ptr<db::view::view_update_generator>& gen,
+    const db::timeout_clock::time_point timeout) {
+
+    // Find the maximum batchlog, initial value is the local batchlog
+    db::view::update_backlog max_backlog = gen->get_storage_proxy().local().get_view_update_backlog();
+
+    for (const frozen_mutation_and_schema& view_update : view_updates) {
```

Comment:
How many of these can we have? `chunked_vector` suggests that his can be potentially large, which makes the run-time of this function a concern, especially w.r.t. stalls.

Reply:
A single batch of updates can contain at most 100 rows worth of updates. This means that the maximum length of this vector is 100, in the case of 100 single-row updates.

100 isn't a lot, the number was chosen to be small enough so that handling all of them at once wouldn't cause stalls.

Potentially we could have a problem if the topology lookup took a long time, but I doubt it, it's in the hot path as it's being called on every update.

Reply:
I don't think the topology lookup is a concern, I was only concerned by the number of elements we have to process here, but if the max is 100 (or that order of magnitude) we should be fine.